hack/generated/pkg/util/patch/patch.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This package is taken from https://github.com/kubernetes-sigs/cluster-api/tree/master/util/patch with slight modifications
+// to suit our use-case
```

Comment:
Could you add what the modifications are? Not heaps of detail but just in terms of what extra we needed.

Reply:
Done